tensorflow/stream_executor/BUILD

```diff
@@ -707,6 +723,30 @@ cc_library(
     ],
 )
 
+cc_library(
+    name = "redzone_allocator",
+    srcs = [] + if_rocm_is_configured([
+        "//tensorflow/stream_executor/rocm:redzone_allocator.cc",
+    ]) + if_cuda_is_configured([
+        "//tensorflow/stream_executor/cuda:redzone_allocator.cc",
```

Comment:
Ideally, could we find a way to build both, and then choose the one to call using an `#ifdef` ?

Reply:
We can continue the discussion [here](https://github.com/tensorflow/tensorflow/pull/30884#discussion_r315300835)